### PR TITLE
Add test cases / fixes for Viewport project. Standardize on array format for lngLat and coordinates.

### PR DESCRIPTION
**Pull request by ibgreen**
This is a breaking change, so will be merged into dev and published as beta3.0.0 release

### CODE REVIEW

**Comment by ibgreen** [comment]
@gnavvy

File: src/layers/base-layer.js [hunk 1]
```diff
@@ -478,14 +482,18 @@ export default class BaseLayer {
   // MAP LAYER FUNCTIONALITY
   setViewport() {
     const {
-      width, height, latitude, longitude, zoom,
+      width, height, latitude, longitude, zoom, pitch, bearing, altitude,
```
Comment (gnavvy) [inline src/layers/base-layer.js]: use pitch=0, bearing=0, altitude=1.5 for backwards compatibility?

File: src/layers/base-layer.js [hunk 2]
```diff
@@ -478,14 +482,18 @@ export default class BaseLayer {
...
     } = this.props;
 
     this.setState({
       viewport: {x: 0, y: 0, width, height},
-      mercator: ViewportMercator({
-        width, height, latitude, longitude, zoom,
+      // mercator: ViewportMercator({
+      //   width, height, latitude, longitude, zoom,
+      //   tileSize: 512
+      // })
```
Comment (gnavvy) [inline src/layers/base-layer.js]: delete above lines?

File: src/viewport/viewport.js [hunk 3]
```diff
@@ -25,7 +25,7 @@ const PI_4 = PI / 4;
 const DEGREES_TO_RADIANS = PI / 180;
 const RADIANS_TO_DEGREES = 180 / PI;
 const TILE_SIZE = 512;
-const WORLD_SCALE = TILE_SIZE / PI_2;
+const WORLD_SCALE = TILE_SIZE / ( 2 * PI);
```
Comment (gnavvy) [inline src/viewport/viewport.js]: extra space before 2

File: src/viewport/viewport.js [hunk 4]
```diff
@@ -74,10 +74,10 @@ export default class Viewport {
     const y = 180 / Math.PI *
       Math.log(Math.tan(Math.PI / 4 + latitude * Math.PI / 360));
 
-    this.centerX0 = (180 + longitude) / 360 * TILE_SIZE;
-    this.centerY0 = (180 - y) / 360 * TILE_SIZE;
-    this.centerX = this.centerX0 * this.scale;
-    this.centerY = this.centerY0 * this.scale;
+    this.center0 = this.projectToWorld([longitude, latitude], 1);
```
Comment (gnavvy) [inline src/viewport/viewport.js]: is this.center0 used elsewhere?

File: src/viewport/viewport.js (hunk 3 above)
Comment (ibgreen) [inline src/viewport/viewport.js]: >extra space before 2

Fixed

File: src/viewport/viewport.js (hunk 4 above)
Comment (ibgreen) [inline src/viewport/viewport.js]: > is this.center0 used elsewhere?

deleted

File: src/layers/base-layer.js (hunk 2 above)
Comment (ibgreen) [inline src/layers/base-layer.js]: > delete above lines?

deleted

File: src/layers/base-layer.js (hunk 1 above)
Comment (ibgreen) [inline src/layers/base-layer.js]: > use pitch=0, bearing=0, altitude=1.5 for backwards compatibility?

Good point, I just don't want to spread out such constants in multiple places (especially the default altitude, which is matched to mapbox's default settings).

I moved the defaults to the Viewport class and made it able to apply default to 'undefined' params (as opposed to just omitted params).

**Pull request merged**